### DISCUSSION TITLE
why the blockid use punctuation+letters not just md5 code?

### DISCUSSION THREAD

**Discussion by OriLiMu** [ScratchAddons/ScratchAddons#7611] [Help] [answered]
I found that the block svg id something like: `"U@)yyHV5hvth+_+mZH)I"`

Why not just use md5 code, which I think more neat.

**Comment by mxmou** [comment, marked as answer]
This isn't directly related to Scratch Addons - the [scratch-blocks repo](https://github.com/scratchfoundation/scratch-blocks) or [Scratch's Advanced Topics forum](https://scratch.mit.edu/discuss/31/) might be a better place for questions like this.

MD5 is a hash function. If it was used to generate block IDs, two identical blocks would have the same ID, which would cause problems. Instead, the IDs are random - the code that generates them is [here](https://github.com/scratchfoundation/scratch-blocks/blob/379db6f/core/utils.js#L610-L633). The same algorithm is also used to generate target (sprite or clone) IDs.
